integration/metrics_server_test.go

```diff
@@ -0,0 +1,65 @@
+package integration_test
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/addon-operator/integration"
+)
+
+func (s *integrationTestSuite) TestMetricsServer() {
+	ctx := context.Background()
+	pod := pod_metricsClient()
+
+	// create the metrics client pod
+	err := integration.Client.Create(ctx, pod)
+	s.Require().NoError(err)
+
+	// wait until Pod is available
+	err = integration.WaitForObject(
+		s.T(), defaultPodAvailabilityTimeout, pod, "to be Ready",
+		func(obj client.Object) (done bool, err error) {
+			p := obj.(*corev1.Pod)
+			for _, podCondition := range p.Status.Conditions {
+				if podCondition.Type == corev1.PodReady && podCondition.Status == corev1.ConditionTrue {
+					return true, nil
+				}
+			}
+			return false, nil
+		})
+	s.Require().NoError(err)
+
+	s.Run("test_https_endpoint", func() {
+		httpsMetricsAddr := "https://addon-operator-metrics.addon-operator.svc:8080/healthz"
+		caCertPath := pod.Spec.Containers[0].VolumeMounts[0].MountPath + "ca.crt"
+
+		command := []string{"curl", "--cacert", caCertPath, httpsMetricsAddr}
+
+		httpsCurlCallResult, _, err := integration.ExecCommandInPod(pod.Namespace, pod.Name, pod.Spec.Containers[0].Name, command)
+		s.Require().NoError(err)
+		s.Assert().Equal("404 page not found", httpsCurlCallResult)
+	})
+
+	s.Run("test_http_endpoint", func() {
+		httpMetricsAddr := "http://addon-operator-metrics.addon-operator.svc:8083/healthz"
```

Comment:
Should we not test `/metrics` endpoint?

Reply:
Sure, we can but this test is just intended to test the existence/reachability of the metrics server and not the actual content of /metrics.
Hence, I chose the /healthz,which despite returning a 404 still proves the existence of the metrics.
Also, the 404 response is deterministic which makes this test more robust (non-err + expected response get validated) but the /metrics endpoint's response is indeterministic hence, validating the expected content/response against /metrics would be pretty complicated and an overkill for this test.